assets/src/application/ui/layout/espressoTable/ResponsiveTable.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import classNames from 'classnames';
+import uuidv4 from 'uuid/v4';
+import { find, isEmpty } from 'lodash'; // to be replaced with ramda
+
+/**
+ * Internal dependencies
+ */
+import Table from './Table';
+import TableHeader from './TableHeader';
+import TableBody from './TableBody';
+import TableFooter from './TableFooter';
+import './style.css';
+
+/**
+ * ResponsiveDataGrid responsive-table
+ * produces a table like structure for displaying tabular data
+ * in a grid that collapses properly on smaller screens
+ *
+ * @param {string} instanceId
+ * @param {Array} headerRows
+ * @param {Array} tableRows
+ * @param {Array} footerRows
+ * @param {Object} metaData
+ * @param {Object} classes
+ * @param {Function} onBeforeDragStart
+ * @param {Function} onDragStart
+ * @param {Function} onDragUpdate
+ * @param {Function} onDragEnd
+ */
+const ResponsiveTable = ({
+	headerRows = [],
+	tableRows = [],
+	footerRows = [],
+	metaData = {},
+	classes = {},
+	onBeforeDragStart,
+	onDragStart,
+	onDragUpdate,
+	onDragEnd,
+	...props
+}) => {
+	const primaryHeader = find(headerRows, 'primary');
+	const instanceId = props.instanceId || uuidv4();
```

Comment:
we can get rid of the `instanceId` since that was coming from a WP HoC,
we also don't **need** something as robust as `uuidv4()` since it just needs to be a unique string within the current DOM and not guaranteed unique across multiple requests with persistence. This could even just be a timestamp